ionoscloud/resource_k8s_node_pool.go

```diff
@@ -266,6 +266,54 @@ func resourceK8sNodePoolUpgradeV0(_ context.Context, state map[string]interface{
 	return state, nil
 }
 
+func getLanResourceData(lansList *schema.Set) []ionoscloud.KubernetesNodePoolLan {
+	lans := make([]ionoscloud.KubernetesNodePoolLan, 0)
+	if lansList.List() != nil {
+		for _, lanItem := range lansList.List() {
+			lanContent := lanItem.(map[string]interface{})
+			lan := ionoscloud.KubernetesNodePoolLan{}
+
+			if lanID, lanIdOk := lanContent["id"].(int); lanIdOk {
+				log.Printf("[INFO] Adding LAN %v to node pool...", lanID)
+				lanID := int32(lanID)
+				lan.Id = &lanID
+			}
+
+			if lanDhcp, lanDhcpOk := lanContent["dhcp"].(bool); lanDhcpOk {
+				log.Printf("[INFO] Adding dhcp %v to node pool...", lanDhcp)
+				lan.Dhcp = &lanDhcp
+			}
+
+			routes := make([]ionoscloud.KubernetesNodePoolLanRoutes, 0)
+
+			if lanRoutes, lanRoutesOk := lanContent["routes"].(*schema.Set); lanRoutesOk {
+				log.Printf("[INFO] Adding routes %v to node pool...", lanRoutes)
+				if lanRoutes.List() != nil {
+					for _, routeItem := range lanRoutes.List() {
+						routeContent := routeItem.(map[string]interface{})
+						route := ionoscloud.KubernetesNodePoolLanRoutes{}
+
+						if routeNetwork, routeNewtworkOk := routeContent["network"].(string); routeNewtworkOk {
+							route.Network = &routeNetwork
```

Comment:
I am not sure how it is used in the rest of the code, and maybe it is a personal preference, but I would recommend the usage of setters: `SetNetwork()`